src/compiler/execution.jl

```diff
@@ -151,12 +151,16 @@ abstract type AbstractKernel{F, TT} end
         end
     end
 
+    pushfirst!(call_t, KernelState)
+    pushfirst!(call_args, :(KernelState(Base.rand(UInt32))))
+
     # finalize types
     call_tt = Base.to_tuple_type(call_t)
 
     quote
         indirect_memory = cl.AbstractMemory[]
-        clcall(kernel.fun, $call_tt, $(call_args...); indirect_memory, call_kwargs...)
+        device_rng = kernel.fun.num_args == $(length(call_args) + 2)
```

Comment:
This is ugly. Could we instead store a flag in the HostKernel object that indicates the RNG is in use? `fun.num_args` requires an API call I'd like to avoid in the kernel launch hot path.